File: BaseBackend/src/main/java/com/common/model/nosql/BaseSoftDeleteModel.java
package com.common.model.nosql;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

/**
 * Lớp model cơ sở hỗ trợ xóa mềm cho các NoSQL MongoDB Document.<br/>
 * Kế thừa từ BaseModel để giữ lại các thông tin kiểm toán (Audit fields).<br/>
 * Created at 19/06/2026
 *
 * @see <a href="../../../../../resources/docs/spring_structure/datasource-guide.md">Datasource and Auditing Guide</a>
 * @author txhoan
 */
@Getter
@Setter
@ToString(callSuper = true)
@SuperBuilder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@FieldDefaults(level = AccessLevel.PRIVATE)
public abstract class BaseSoftDeleteModel extends BaseModel {

    @lombok.Builder.Default
    boolean deleted = false;

    LocalDateTime deletedAt;

    String deletedBy;
}

File: be/.Net/Swe.Common/Base/BaseSoftDeleteModel.cs
using Swe.Common.Attributes;

namespace Swe.Common.Base;

public abstract class BaseSoftDeleteModel : BaseModel
{
    [ConfigColumn("deleted")]
    public bool Deleted { get; set; } = false;

    [ConfigColumn("deleted_at")]
    public DateTime? DeletedAt { get; set; }

    [ConfigColumn("deleted_by")]
    public string? DeletedBy { get; set; }
}
